src/Altitude.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {IncrementalTreeData, IncrementalBinaryTree} from "./IncrementalBinaryTree.sol";
+import {Ownable2Step} from "../lib/openzeppelin-contracts/contracts/access/Ownable2Step.sol";
+import {ECDSA} from "../lib/openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
+import {PoseidonT3} from "../lib/poseidon-solidity/contracts/PoseidonT3.sol";
+import {PoseidonT5} from "../lib/poseidon-solidity/contracts/PoseidonT5.sol";
+
+struct Force {
+    uint hash;
+    uint time;
+}
+
+contract Altitude is Ownable2Step {
+    string internal constant ERROR_MINT_EMPTY = "E1";
+    string internal constant ERROR_FORCE_NO_COIN = "E2";
+    string internal constant ERROR_SLASH_NO_FORCE = "E3";
+    string internal constant ERROR_SLASH_NOT_OLD = "E3";
+    using IncrementalBinaryTree for IncrementalTreeData;
+
+    uint internal constant ZERO = 0;
+    uint internal constant DEPTH = 20;
+    uint internal constant FORCE_WAIT = 1 days;
+
+    IncrementalTreeData public mintTree;
+    uint public stateRoot;
+
+    Force[] public forcedInclusions;
+
+    uint maxCoin = 1;
+
+    constructor(address _owner) {
+        _transferOwnership(_owner);
+        mintTree.init(DEPTH, ZERO);
+    }
+
+    function mint(address receiver) external payable returns (uint root) {
+        require(msg.value > 0, ERROR_MINT_EMPTY);
+        root = mintTree.insert({
+            leaf: PoseidonT3.hash([uint(uint160(receiver)), msg.value])
```

Comment:
Does `[uint(uint160(receiver))` make sure that it's a 160 bit number, but it's still ultimately taking up the memory of a 256 bit number?